batchflow/models/tf/nn/train.py

```diff
@@ -0,0 +1,74 @@
+""" Helpers for training """
+from math import pi
+
+import tensorflow as tf
+from tensorflow.python.framework import ops # pylint: disable=no-name-in-module
+from tensorflow.python.ops import math_ops # pylint: disable=no-name-in-module
+from tensorflow.math import sin, asin, floor # pylint: disable=import-error
+
+
+def piecewise_constant(global_step, *args, **kwargs):
+    """ Constant learning rate decay (uses global_step param instead of x) """
+    return tf.train.piecewise_constant(global_step, *args, **kwargs)
+
+def cyclic_learning_rate(learning_rate, global_step, max_lr=0.1, step_size=10, mode='triangular', name=None):
+    """Applies cyclic learning rate (CLR).
+    https://arxiv.org/pdf/1506.01186.pdf.
+
+    This function varies the learning rate between the
+    minimum (learning_rate) and the maximum (max_lr).
+    It returns the cyclic learning rate.
+
+    Args:
+    learning_rate: A scalar `float32` or `float64` `Tensor` or a
+        Python number.  The initial learning rate which is the lower bound
+        of the cycle.
+    global_step: A scalar `int32` or `int64` `Tensor` or a Python number.
+        Global step to use for the cyclic computation.  Must not be negative.
+    max_lr:  A scalar. The maximum learning rate boundary.
+        Default 0.1.
+    step_size: A scalar. The number of iterations in half a cycle.
+        Default 10.
+    mode:
+        If 'sin' or 'sine' or 'sine wave':
+            Learning rate changes as a sine wave, starting
+            from (max_lr-learning_rate)/2 then decreasing to `learning_rate`.
+        If 'triangular' or 'triangular wave' or 'zigzag':
+            Default, linearly increasing then linearly decreasing the
+            learning rate at each cycle. Learning rate starting
+            from (max_lr-learning_rate)/2 then decreasing to `learning_rate`.
+        If 'sawtooth' or 'saw' or 'sawtooth wave' or 'saw wave':
+            Linearly increasing to `max_lr` then again increasing from `learning_rate`
```

Comment:
how this differs from 'trianguar'?

Reply:
In `saw` mode learning rate linearly increases from `learning rate` to `max_lr` and then sharply drops to `learning rate`.

Reply:
Add this to docs